trin-core/src/utp/stream.rs

```diff
@@ -875,7 +877,9 @@ impl UtpStream {
 
     /// Handle uTP socket timeout
     async fn handle_receive_timeout(&mut self) -> anyhow::Result<()> {
-        self.congestion_timeout *= 2;
+        self.congestion_timeout = self
+            .congestion_timeout
+            .saturating_add(MIN_CONGESTION_TIMEOUT);
```

Comment:
This change fixes the overflow, but it removes the exponential backoff.

Why not

```rust
self.congestion_timeout = self.congestion_timeout.saturating_mul(2);
```

Reply:
After reviewing this, I don't think we need to double the congestion timeout every time when we resend a single packet. Increasing the congestion timeout with `MIN_CONGESTION_TIMEOUT` (500ms) seems more reasonable to me.

Reply:
Based on [spec](https://www.bittorrent.org/beps/bep_0029.html): "For every packet consecutive subsequent packet that times out, the timeout is doubled."

I'm not sure what the behavior of the reference implementation is though.

I do think that exponential backoff is generally more effective than linear backoff, so I would be in favor of keeping the doubling. You seem to have some reason to favor the linear backoff, but I'm not sure what that is. Can you elaborate?

Reply:
Didn't know this was in the specs, let's keep it close to the specs then, changing back to `saturating_mul(2)`.